packages/dotlottie-js/src/common/utils.ts

```diff
@@ -830,3 +830,79 @@ export async function getTheme(
 
   return strFromU8(unzippedTheme, false);
 }
+
+/**
+ * Retrieves the state machines from the given DotLottie object.
+ *
+ * @remarks
+ * This function accepts a DotLottie object as a Uint8Array and an optional filter function to refine the extraction of state machines.
+ * It returns a Promise that resolves to a record containing the state machines mapped by their ID.
+ *
+ * @param dotLottie - The DotLottie object containing the state machines.
+ * @param filter - An optional function to filter the files to be unzipped.
+ * @returns A Promise that resolves to a record containing the state machines mapped by their ID.
+ *
+ * @example
+ * ```typescript
+ * const dotLottie = new Uint8Array(...);
+ * const machines = await getStateMachines(dotLottie);
+ * ```
+ */
+export async function getStateMachines(
+  dotLottie: Uint8Array,
+  filter?: UnzipFileFilter,
+): Promise<Record<string, string>> {
+  const statesMap: Record<string, string> = {};
+
+  const unzippedStates = await unzipDotLottie(
+    dotLottie,
+    (file) => file.name.startsWith('states/') && (!filter || filter(file)),
+  );
+
+  for (const statePath in unzippedStates) {
+    const data = unzippedStates[statePath];
+
+    if (data instanceof Uint8Array) {
+      const themeId = statePath.replace('states/', '').replace('.json', '');
+
+      statesMap[themeId] = strFromU8(data, false);
+    }
+  }
+
+  return statesMap;
+}
+
+/**
+ * Retrieves a specific state machine by ID from the given DotLottie object.
+ *
+ * @remarks
+ * This function accepts a DotLottie object as a Uint8Array, the state ID to retrieve, and an optional filter function.
+ * It returns a Promise that resolves to the state machine as a string or `undefined` if not found.
+ *
+ * @param dotLottie - The DotLottie object containing the theme.
+ * @param stateMachineId - The ID of the state machine to retrieve.
+ * @param filter - An optional function to filter the files to be unzipped.
+ * @returns A Promise that resolves to the state machine as a string or `undefined` if not found.
+ *
+ * @example
+ * ```typescript
+ * const dotLottie = new Uint8Array(...);
+ * const stateMachineId = 'walk';
+ * const stateMachine = await getState(dotLottie, stateMachineId);
+ * ```
+ */
+export async function getStateMachine(
+  dotLottie: Uint8Array,
+  stateMachineId: string,
+  filter?: UnzipFileFilter,
+): Promise<string | undefined> {
```

Comment:
@theashraf I removed the StateMachine class because it forced us to have an extra 'state' in the StateMachine:

pseudocode
```
Statemachine {
descriptor: StateMachineDescriptor
states: DotLottieStates
}
```

Then in usage it became:

```
{
descriptor: {..},
states: {
  walkState {
  },
  ...
}
```

This is because in the DotLottieStates I have:

```
  [key: string]: StateSettings;
```

Which doesn't allow us to define something like this which I wanted in the first place:

```
State {
descriptor: StateDescriptor,
[key: string]: StateSettings;
```

I removed it and now the DotLottieState class serves to assemble them.

For this method would this be better?

```
export async function getStateMachine(
  dotLottie: Uint8Array,
  stateMachineId: string,
  filter?: UnzipFileFilter,
): Promise<LottieStateMachine | undefined> {
  const stateMachineFilename = `states/${stateMachineId}.json`;

  const unzippedStateMachine = await unzipDotLottieFile(dotLottie, stateMachineFilename, filter);

  if (typeof unzippedStateMachine === 'undefined') {
    return undefined;
  }

  const stateMachine = JSON.parse(strFromU8(unzippedStateMachine, false)) as LottieStateMachine;

  return stateMachine;
}
```

Reply:
looks good